MantidPlot/src/Mantid/InstrumentWidget/InstrumentWindow.cpp

```diff
@@ -839,8 +60,8 @@ void InstrumentWindow::preDeleteHandle(
   Mantid::API::IPeaksWorkspace_sptr pws =
       boost::dynamic_pointer_cast<Mantid::API::IPeaksWorkspace>(workspace_ptr);
   if (pws) {
-    getSurface()->deletePeaksWorkspace(pws);
-    updateInstrumentView();
+    m_instrumentWidget->getSurface()->deletePeaksWorkspace(pws);
+    m_instrumentWidget->updateInstrumentView();
```

Comment:
I think the `deletePeaksWorkspace` adjustment suggested above should also call updateInstrumentView for the client in this case. Can we get away with automatically updating the `InstrumentViewWidget` when we know that we have to in other cases as well?